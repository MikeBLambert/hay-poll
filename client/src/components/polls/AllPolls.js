import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';

export default class AllPolls extends PureComponent {

  static propTypes = {
    polls: PropTypes.array.isRequired
  }

  componentDidMount() {
    const { fetchPolls } = this.props;
    fetchPolls();
  }



  render() {
    const { polls } = this.props;
    const pollsList = polls.map(poll => {
      return (
        <div key={poll._id}>
          <Link to={ROUTES.POLL_DETAIL.linkTo(poll._id)}>{poll.issue}</Link>
        </div>
      );
    });

    return (
      <div>
        <ul>{pollsList}</ul>
      </div>
    );
  }
}
