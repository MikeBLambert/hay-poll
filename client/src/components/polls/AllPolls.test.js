import React from 'react';
import { shallow } from 'enzyme';
import AllPolls from './AllPolls';

describe('AllPolls component', () => {
  it('renders a component', () => {
    const polls = [{ issue: 'What is your favorite TV Show?', options: ['GOT', 'Sopranos', 'Halt and Catch Fire'] }];
    const wrapper = shallow(<AllPolls polls={polls} />);
    expect(wrapper).toMatchSnapshot();
  });
});
