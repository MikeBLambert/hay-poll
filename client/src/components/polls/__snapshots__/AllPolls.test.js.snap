// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllPolls component renders a component 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      What is your favorite TV Show?
    </li>
  </ul>
</div>
`;
